src/auth/executableresponse.ts

```diff
@@ -0,0 +1,195 @@
+  /**
+   * The version of the Executable response. Only version 1 is currently supported.
+   */
+  readonly version: number;
+  /**
+   * Whether the executable ran successfully.
+   */
+  readonly success: boolean;
+  /**
+   * Epoch time for expiration of the token in seconds.
+   */
+  readonly expirationTime?: number;
+  /**
+   * Type of subject token in the response, currently supported values are:
+   * urn:ietf:params:oauth:token-type:saml2
+   * urn:ietf:params:oauth:token-type:id_token
+   * urn:ietf:params:oauth:token-type:jwt
+   */
+  readonly tokenType?: string;
+  /**
+   * Error code from executable.
+   */
+  readonly errorCode?: string;
+  /**
+   * Error message from executable.
+   */
+  readonly errorMessage?: string;
+  /**
+   * Subject token from executable, format depends on tokenType.
+   */
+  readonly subjectToken?: string;
+
+  /**
+   * Instantiates an ExecutableResponse instance using the provided JSON object
+   * from the output of the executable.
+   * @param responseJson Response from a 3rd party executable, loaded from a
+   * run of the executable or a cached output file.
+   */
+  constructor(responseJson: ExecutableResponseJson) {
+    // Check that always required fields exist in the json response.
+    if (!responseJson.version) {
+      throw Error("Executable response must contain a 'version' field.");
+    }
+    if (responseJson.success === undefined) {
+      throw Error("Executable response must contain a 'success' field.");
+    }
+
+    this.version = responseJson.version;
+    this.success = responseJson.success;
+
+    // Validate required fields for a successful response.
+    if (this.success) {
+      if (!responseJson.expiration_time) {
+        throw Error(
+          "Executable response must contain an 'expiration_time' field when successful."
+        );
+      }
+      if (!responseJson.token_type) {
+        throw Error(
+          "Executable response must contain a 'token_type' field when successful."
+        );
+      }
+      this.expirationTime = responseJson.expiration_time;
+      this.tokenType = responseJson.token_type;
+
+      // Validate token type field.
+      if (
+        this.tokenType !== SAML_SUBJECT_TOKEN_TYPE &&
+        this.tokenType !== OIDC_SUBJECT_TOKEN_TYPE1 &&
+        this.tokenType !== OIDC_SUBJECT_TOKEN_TYPE2
+      ) {
+        throw new Error(
+          "Executable response must contain a 'token_type' field when successful " +
+            `and it must be one of ${OIDC_SUBJECT_TOKEN_TYPE1}, ${OIDC_SUBJECT_TOKEN_TYPE2}, or ${SAML_SUBJECT_TOKEN_TYPE}.`
+        );
+      }
+
+      // Validate subject token.
+      if (this.tokenType === SAML_SUBJECT_TOKEN_TYPE) {
+        if (!responseJson.saml_response) {
+          throw Error(
+            `Executable response must contain a 'saml_response' field when token_type=${SAML_SUBJECT_TOKEN_TYPE}.`
+          );
+        }
+        this.subjectToken = responseJson.saml_response;
+      } else if (
```

Comment:
This else if condition is redundant given that you validated the token type above